### PR TITLE
fix: [IA-566] Fix entrypoints on error and attempt screens

### PR DESCRIPTION
**Pull request by debiff**
## Short description
This PR manage the `assistanceTools` in the transaction error screen and in the detail of payment and credit card attempt.

## List of changes proposed in this pull request
- Extracted the `canShowHelp` function
- Managed the assistance tools in `TransactionErrorScreen`
- Managed the assistance tools in `PaymentHistoryDetailsScreen`
- Managed the assistance tools in `CreditCardOnboardingAttemptDetailScreen`

https://user-images.githubusercontent.com/11773070/145787436-c5eac70c-4922-4eeb-ae19-a0236b012466.mp4

https://user-images.githubusercontent.com/11773070/145787467-78f18326-801e-4137-b215-f558162d8b80.mp4

https://user-images.githubusercontent.com/11773070/145787484-a6bacef9-fcf6-4556-9f8a-2ea19742684b.mp4

## How to test
Fail to add a credit card or fail a payment and modify the assitanceTool in the [dev-server](https://github.com/pagopa/io-dev-api-server/blob/bd71dd1d55cffb23c17e34bcb8979e5bf2266b05/src/payloads/backend.ts#L198) to try the different combinations.

### CODE REVIEW

**Review by thisisjp — APPROVED** [review]
LGTM @debiff ideally users shouldn't select any category when asking help using these entry points (we already know it's a payment-related issue). Can we track this improvement in our backlog? 🙏

**Comment by Undermaken** [comment]
@debiff could you solve conflicts?

**Comment by debiff** [comment]
> LGTM @debiff ideally users shouldn't select any category when asking help using these entry points (we already know it's a payment-related issue). Can we track this improvement in our backlog? 🙏

Yes, this is not a great problem, we can do it! 💪

**Comment by debiff** [comment]
> @debiff could you solve conflicts?

Sure, done!

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/screens/wallet/creditCardOnboardingAttempts/CreditCardOnboardingAttemptDetailScreen.tsx [hunk 1]
```diff
@@ -65,14 +71,31 @@ const instabugTag = "credit-card-support";
...
+    dispatch(zendeskSupportStart({ startingRoute: "n/a" }));
+  };
+
+  const handleAskAssistance = () => {
+    switch (choosenTool) {
+      case ToolEnum.instabug:
+        instabugLogAndOpenReport();
+        break;
+      case ToolEnum.zendesk:
+        zendeskAssistanceLogAndStart();
```
[the remark this answers is not in the document]
Comment (debiff) [reply ts/screens/wallet/creditCardOnboardingAttempts/CreditCardOnboardingAttemptDetailScreen.tsx:96]: Added in [db6bde7](https://github.com/pagopa/io-app/pull/3573/commits/db6bde7f92699f5556ac4fc1c42dbabd879a3eb6)

File: ts/screens/wallet/PaymentHistoryDetailsScreen.tsx [hunk 2]
```diff
@@ -99,6 +108,26 @@ class PaymentHistoryDetailsScreen extends React.Component<Props> {
...
+    this.props.assistanceToolConfig
+  );
+
+  private handleAskAssistance = () => {
+    switch (this.choosenTool) {
+      case ToolEnum.instabug:
+        this.instabugLogAndOpenReport();
+        break;
+      case ToolEnum.zendesk:
+        this.zendeskAssistanceLogAndStart();
```
[the remark this answers is not in the document]
Comment (debiff) [reply ts/screens/wallet/PaymentHistoryDetailsScreen.tsx:127]: Added in [db6bde7](https://github.com/pagopa/io-app/pull/3573/commits/db6bde7f92699f5556ac4fc1c42dbabd879a3eb6)

**Review by Undermaken — APPROVED** [review]

**Pull request merged**